mergin/merginproject.py

```diff
@@ -154,10 +158,18 @@ def workspace_name(self) -> str:
         return full_name[:slash_index]
 
     def project_id(self) -> str:
-        """Returns ID of the project (UUID using 8-4-4-4-12 formatting without braces)"""
+        """Returns ID of the project (UUID using 8-4-4-4-12 formatting without braces)
+
+        Raises ClientError if project id is not present in the project metadata.
+        """
         if self._metadata is None:
             self._read_metadata()
-        return self._metadata["project_id"]
+
+        # "id" or "project_id" may not exist in projects downloaded with old client version
+        if "id" not in self._metadata and "project_id" not in self._metadata:
+            raise ClientError("Missed project id metadata. Please re-download your project.")
```

Comment:
I would suggest to re-phrase this slightly: `The project directory has been created with an old version of the Mergin Maps client. Please delete the project directory and re-download the project.`

Reply:
Changed.